Slim/Web/Pages/Search.pm

```diff
@@ -508,7 +509,11 @@ sub _initActiveRoles {
 		$params->{'search'}->{'contributor_namesearch'}->{'active' . $_} = 1 if $params->{'search.contributor_namesearch.active' . $_};
 	}
 
-	$params->{'search'}->{'contributor_namesearch'} = { map { ('active' . $_) => 1 } @{ Slim::Schema->artistOnlyRoles } } unless keys %{$params->{'search'}->{'contributor_namesearch'}};
+	my @udr;
+	foreach (keys %{$serverPrefs->get('userDefinedRoles')}) {
+		push @udr, $_ if $serverPrefs->get('userDefinedRoles')->{$_}->{'include'};
+	}
```

Comment:
Doesn't belong here.

Reply:
...and we wouldn't even need this if the condition on the next line wasn't met. Please wrap this all in a regular `if {}` block. Should make the next line more readable, too.

In general we should try to avoid in-line conditions if the line becomes overly long. The condition can easily be hidden outside the screen. If you still prefer the in-line condition you could at least format the map to make it use multiple lines. But in this case we actually don't need the `@udr` unless the condition is met. Therefore just go regular `if`.

Reply:
Of course! in my defence all I did was add the parameter `@udr` to the existing code (see old line 511).  I'll make the change.